src/fetchData.js

```diff
@@ -0,0 +1,22 @@
+import { display } from './domStuff';
+
+const url = 'https://api.openweathermap.org/data/2.5/weather?q=';
+const key = '9bf5614e989b618c0574d6238d81db95';
+
+// const key = '9bf5614e989b618c0574d6238d81db95';
+
+
+function basic(city) {
```

Comment:
We want our basic function to simply return the result of fetching from the api. We don't want to include here anything that we do to it afterwards. This may not be the best place where we call the `display` function. 

```javascript
const basic = async (city) => {
  const baseUrl = 'https://api.openweathermap.org/data/2.5/weather?q=';
  const key = '9bf5614e989b618c0574d6238d81db95';
  const path = `${baseUrl}${city}&units=metric&APPID=${key}`;
  const data =  await fetch(path, {mode: 'cors'})
                         .then(res => {
                                      if(res.status === 200) return res.json();
                                      else return Promise.reject("Cannot fetch data for city");
                           });
   return data;
}
```
We need the `if else` in basic because we want to make sure we catch the scenario where the user inputs a city whose data doesn't exist (for instance he/she types "King's Landing" in the input box).

And then we can simply export this function to be processed in the `domStuff` file

```
export default basic;
```

And then in the `domStuff.js` file `addEventListener` function we can use it as such:

```javascript

basic(city)
 .then(data => display(data))
 .catch(err => // do something with the error that gets returned (the return statement in the else part of basic))
```
We need error handling to be handled here. basic returns a promise that may not resolve (i.e, network problems, or perhaps the city cannot be found). So we want to make sure our user will be notified. As it is right now, if I search for a city that doesn't exist it just gives me undefined in the display.

Reply:
Thanks for your help @rvvergara